tests/AggregationBlogPost/article.md

```diff
@@ -0,0 +1,713 @@
+
+Now that we know about these value holder objects, we still need to make sure
+the server knows what we're talking about. When you call `Collection::aggregate`
+with the pipeline you built, what happens internally to it? Here, a series of
+encoders springs into action. We use a single entry point, the `BuilderEncoder`
+class. This class contains multiple encoders that are able to handle all
+pipeline stages, operators, and accumulators and transform them into their BSON
+representations.
+
+This allows us to keep the logic customizable. For example, the Doctrine MongoDB
+ODM allows users to specify different names for fields in the database. In turn,
+it needs to convert the name of a property in the mapped class to the name of
+the field in the database. Such a feature could easily be built by creating a
+custom encoder for all `fieldPath` expressions, and changing the field path
+accordingly.
+
+When creating a `MongoDB\Client` instance, you can now pass an additional
+`builderEncoder` option in the `$driverOptions` argument. This specifies the
+encoder used to encode aggregation pipelines, but also query objects. All
+`Database` and `Client` instances inherit this value from the client, but you
+can override it through the options when fetching such an instance. This allows
+you to have your custom logic applied whenever pipelines or queries are encoded
+for the server.
+
+With factories, value holders, and encoders, we wanted to ensure that creating
+the builder does not turn into a repetitive chore. As you can imagine, many
+operators will mostly consist of the same logic, resulting in tons of code
+duplication. To make matters worse, every new server version adds some new
+operators or even stages, so we wanted to make sure that we can easily expand
+the builder.
+
+We could try to rely on generative AI to help us with this, but this only goes
+so far. Instead, we leverage code generation to make the task easier. All
+factories, value holders, and encoders are generated from a configuration. When
+a new operator is introduced, we create a config file with all of its details:
+input types, what the operator resolves to, documentation for parameters, and
+even the examples from the documentation are included. We then run the
+generator, and are given all code necessary to use the operator.
+
+As if that wasn't good enough, the generator also takes the examples we added
+before and adds them to the tests. In these tests, we manually write the builder
+code that would generate the pipeline or expression from the example. This
+ensures that the generated logic behaves as we expect, is protected from
+regression should we make any changes to the generator, and it allows us to use
+the builder and feel what it's like. To top it all off, we could add this code
+to the server documentation, similar to how we add other language-specific code
+snippets. We're not quite there yet, but we'd love to include this in the
+documentation.
```

Comment:
> Ultimately, we love to add this builder code to the server documentation, similar to how we add other language-specific code snippets, but we're not quite there yet.

See if that reads better. I don't think this idea warrants more than a single sentence.